components/Modal/Modal.tsx

```diff
@@ -45,6 +45,21 @@ export const Modal: React.FC<IModalComponentProps> = withMooskinContext((props)
 		}
 
 		return React.Children.map(children, (child, i) => {
+			// on press esc button close modal
+			const escButtonHandler = (event:any) => {				
+				if(event.keyCode === 27){
+					if (React.isValidElement<IModalOverlayComponentProps>(child) && child.type === ModalCloseButton) {
```

Comment:
No need to check if there is a close icon component, the modal should close non the less.